pkg/engine/context/evaluate.go

```diff
@@ -40,29 +40,36 @@ func (ctx *context) Query(query string) (interface{}, error) {
 
 func (ctx *context) loadDeferred(query string) error {
 	loaders := ctx.getMatchingLoaders(query)
-	for _, l := range loaders {
-		if err := l(); err != nil {
+	for _, loader := range loaders {
+		err := ctx.evaluateLoader(loader)
+		if err != nil {
 			return err
 		}
 	}
-
 	return nil
 }
 
-func (ctx *context) getMatchingLoaders(query string) []DeferredLoader {
+func (ctx *context) getMatchingLoaders(query string) []string {
 	ctx.deferred.mutex.Lock()
 	defer ctx.deferred.mutex.Unlock()
-	var matchingLoaders []DeferredLoader
-	for name, deferredLoader := range ctx.deferred.loaders {
+	var matchingLoaders []string
+	for name := range ctx.deferred.loaders {
 		if strings.Contains(query, name) {
-			matchingLoaders = append(matchingLoaders, deferredLoader)
-			delete(ctx.deferred.loaders, name)
+			matchingLoaders = append(matchingLoaders, name)
 		}
 	}
-
 	return matchingLoaders
 }
 
+func (ctx *context) evaluateLoader(name string) error {
+	loader, ok := ctx.deferred.loaders[name]
+	if !ok {
```

Comment:
What's the reason for not removing the loader from context if not `ok`? Maybe I am missing a part of the logic here.

Reply:
Oh wait - it's not ok if it doesn't exist. So it makes sense.